third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -376,6 +376,171 @@ createStreamOps(const LoadToInfoMap &loadToInfo, scf::ForOp &forOp,
   return loadToStreamOp;
 }
 
+static bool isCoalesced(RankedTensorType loadType,
```

Comment:
Can we add some documetation to mention the relationship of these parameters and how they involve to compute coalescing? Something like "Returns true if given a global load with `loadType`, loading instead with `llEnc` maintains at least the same coalescing/vectorization with same amount of load ops."